back-end/tests/endpoints.test.js

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable camelcase */
+/* eslint-disable no-undef */
+const app = require('../server.js');
+const supertest = require('supertest');
+const request = supertest(app);
+
+describe('badges', () => {
+  it('should return an array of objects', async done => {
+    const response = await request.get('/badges');
+    expect(response.status).toBe(200);
+    expect(response.body.length).toBeGreaterThan(0);
+    expect(response.body[0]).toMatchObject({
+      id: 1,
+      name: "A stealthy aquaintance",
+      requirement: "Complete 1 Rogue Quest",
+      int_requirement: 1,
+      criteria_type: "quest",
+      class_id: 1
+    });
+    done();
+  });
```

Comment:
All the tests in this file could be made more readable if they were not using the `done()` callback.

For instance:

```js
it('should return an array of objects', async () => {
    const response = await request.get('/badges');
    expect(response.status).toBe(200);
    expect(response.body.length).toBeGreaterThan(0);
    expect(response.body[0]).toMatchObject({
      id: 1,
      name: "A stealthy acquaintance",
      requirement: "Complete 1 Rogue Quest",
      int_requirement: 1,
      criteria_type: "quest",
      class_id: 1
    });
});
```

See [Jest documentation](https://jestjs.io/docs/en/asynchronous#asyncawait).